From a471ccfd5b2c7810506aac71d4eb3616a3fb18f9 Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Wed, 4 Dec 2024 01:43:20 +0100
Subject: [PATCH] arm64: dts: qcom: Add missing clock for nsscc from pcs uniphy

Add missing clock for nsscc from PCS UNIPHY to scale frequency of each
clock based on the requested PHY.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 arch/arm64/boot/dts/qcom/ipq9574.dtsi | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

--- a/arch/arm64/boot/dts/qcom/ipq9574.dtsi
+++ b/arch/arm64/boot/dts/qcom/ipq9574.dtsi
@@ -14,6 +14,7 @@
 #include <dt-bindings/reset/qcom,ipq9574-gcc.h>
 #include <dt-bindings/clock/qcom,ipq9574-nsscc.h>
 #include <dt-bindings/reset/qcom,ipq9574-nsscc.h>
+#include <dt-bindings/clock/qcom,ipq-pcs-uniphy.h>
 #include <dt-bindings/thermal/thermal.h>
 
 / {
@@ -832,12 +833,12 @@
 				 <&cmn_pll NSS_1200MHZ_CLK>,
 				 <&cmn_pll PPE_353MHZ_CLK>,
 				 <&gcc GPLL0_OUT_AUX>,
-				 <0>,
-				 <0>,
-				 <0>,
-				 <0>,
-				 <0>,
-				 <0>,
+				 <&pcsuniphy0 UNIPHY_NSS_RX_CLK>,
+				 <&pcsuniphy0 UNIPHY_NSS_TX_CLK>,
+				 <&pcsuniphy1 UNIPHY_NSS_RX_CLK>,
+				 <&pcsuniphy1 UNIPHY_NSS_TX_CLK>,
+				 <&pcsuniphy2 UNIPHY_NSS_RX_CLK>,
+				 <&pcsuniphy2 UNIPHY_NSS_TX_CLK>,
 				 <&gcc GCC_NSSCC_CLK>;
 			#clock-cells = <1>;
 			#reset-cells = <1>;
